storage/imx6ull-flash/flashsrv.c

```diff
@@ -767,20 +782,138 @@ static void daemonize(void)
 }
 
 
+static int parse_opts(int argc, char **argv)
+{
+	int c, rootfs_first = -1, rootfs_second = -1;
+	char *p, *part_name;
+	size_t part_start, part_size;
+
+	while ((c = getopt(argc, argv, "r:p:")) != -1) {
+		switch (c) {
+			case 'r': /* fs_name:rootfs_part_id[:secondary_rootfs_part_id] */
+
+				flashsrv_common.rootfs_fsname = optarg;
+				p = strchr(optarg, ':');
+				if (!p) {
+					LOG_ERROR("missing rootfs filesystem name");
+					return -1;
+				}
+
+				*p++ = '\0';
+
+				errno = 0;
+				rootfs_first = strtol(p, &p, 10);
+				if (*p++ == ':') {
+					rootfs_second = strtol(p, &p, 10);
+				}
+
+				break;
+
+			case 'p': /* start:size[:name] */
+				/* TODO: switch to positional arguments to highlight partition order matter */
+
+				errno = 0;
+				part_start = strtol(optarg, &p, 10);
+				if (*p++ != ':') {
```

Comment:
`char *p` might be uninitialized in this case

Reply:
how? if the entire input string is valid, `strtol` will set `*endptr` as a pointer to `\0`

Reply:
Yes, didn't notice p in strtol